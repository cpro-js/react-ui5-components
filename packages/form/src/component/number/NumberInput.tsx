import { InputDomRef } from "@ui5/webcomponents-react";
import { FC, HTMLAttributes, forwardRef, useContext } from "react";

import { BaseNumberInput } from "./BaseNumberInput";
import { NumberContext } from "./context/NumberContext";
import type {
  CommonNumberInputProps,
  NumberDisplayConfig,
  NumberInputConfig,
} from "./NumberModel";

// pick only those props which we do care about
type SharedHtmlProps = Pick<
  HTMLAttributes<HTMLElement>,
  | "style"
  | "className"
  | "id"
  | "placeholder"
  | "title"
  | "onKeyDown"
  | "onBlur"
  | "onFocus"
  | "onPaste"
  | "onMouseOver"
  | "onMouseOut"
  | "onMouseEnter"
  | "onMouseLeave"
  | "onMouseMove"
>;

export type NumberInputProps = SharedHtmlProps &
  NumberDisplayConfig &
  NumberInputConfig &
  Pick<
    CommonNumberInputProps,
    | "value"
    | "onChange"
    | "onValue"
    | "onKeyUp"
    | "locale"
    | "showNumberWarningMessages"
    | "getNumberWarningMessage"
    | "children"
    | "icon"
    | "valueStateMessage"
    | "disabled"
    | "name"
    | "noTypeahead"
    | "placeholder"
    | "readonly"
    | "required"
    | "showClearIcon"
    | "valueState"
  >;

/** `NumberInput` as a wrapper for
 *  <a href="https://sap.github.io/ui5-webcomponents-react/?path=/docs/inputs-input--docs" target="_blank">Ui5 Input</a>
 *  modified to only accept numbers */
export const NumberInput = forwardRef<InputDomRef, NumberInputProps>(
  (props, forwardedRef) => {
    // extract currency specific fields from context
    const { currency, showCurrency, ...contextProps } =
      useContext(NumberContext);

    return <BaseNumberInput {...contextProps} {...props} ref={forwardedRef} />;
  }
);
