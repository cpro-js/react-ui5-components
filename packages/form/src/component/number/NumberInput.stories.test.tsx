import { StoryFn } from "@storybook/react";
import { composeStories } from "@storybook/testing-react";

import { renderWithDefine } from "../../test/render";
import * as stories from "./NumberInput.stories";

const testCases = Object.values(composeStories(stories)).map(
  (StoryFn: StoryFn) => [StoryFn.storyName!, StoryFn]
);
// Batch snapshot testing
test.skip.each(testCases)("Renders %s story", async (_storyName, StoryFn) => {
  const tree = await renderWithDefine(<StoryFn />);
  expect(tree.container).toMatchSnapshot();
});
