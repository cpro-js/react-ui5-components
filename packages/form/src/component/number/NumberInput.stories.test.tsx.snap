// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders Display Config: MaxFractionDigits = 1 story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="123456.8"
    value-state="None"
  />
</div>
`;

exports[`Renders Display Config: MaxSignificantDigits = 3 story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="123000"
    value-state="None"
  />
</div>
`;

exports[`Renders Display Config: MinFractionDigits = 6 story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="1.234000"
    value-state="None"
  />
</div>
`;

exports[`Renders Display Config: MinIntegerDigits = 3 story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="001"
    value-state="None"
  />
</div>
`;

exports[`Renders Display Config: MinSignificantDigits = 3 story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="1.00"
    value-state="None"
  />
</div>
`;

exports[`Renders Input Config: Integers Only = maxFractionDigits = 0 story 1`] = `
<div>
  <ui5-input
    inputmode="numeric"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="123457"
    value-state="None"
  />
</div>
`;

exports[`Renders Input Config: MaxFractionDigits = 1 story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="123456.8"
    value-state="None"
  />
</div>
`;

exports[`Renders Input Config: MaxValue = 999 story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="999"
    value-state="None"
  />
</div>
`;

exports[`Renders Input Config: MinValue = -3 story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="123456.789"
    value-state="None"
  />
</div>
`;

exports[`Renders Input Config: MinValue = 0 story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="123456.789"
    value-state="None"
  />
</div>
`;

exports[`Renders Prefilled story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="123456.789"
    value-state="None"
  />
</div>
`;

exports[`Renders Standard story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value=""
    value-state="None"
  />
</div>
`;

exports[`Renders WithGrouping story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="123,456.789"
    value-state="None"
  />
</div>
`;

exports[`Renders WithIcon story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value=""
    value-state="None"
  >
    <span
      slot="icon"
    >
      EUR
    </span>
  </ui5-input>
</div>
`;

exports[`Renders WithLocaleDe story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="123.456,789"
    value-state="None"
  />
</div>
`;

exports[`Renders WithLocaleIn story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="1,23,456.789"
    value-state="None"
  />
</div>
`;

exports[`Renders WithLocalePl story 1`] = `
<div>
  <ui5-input
    inputmode="decimal"
    maxlength="18"
    type="Text"
    ui5-input=""
    value="123Â 456,789"
    value-state="None"
  />
</div>
`;
